packages/@sanity/validation/src/validateDocument.js

```diff
@@ -131,7 +131,13 @@ function validatePrimitive(item, type, path, options) {
 
   const results = type.validation.map(rule =>
     rule
-      .validate(item, {parent: options.parent, document: options.document, path})
+      .validate(item, {
+        parent: options.parent,
+        document: options.document,
+        path,
+        fieldOptions: options.type && options.type.options,
+        typeName: options.type && options.type.name
```

Comment:
Since it's now two fields from the `type` we are sending in here, would it make more sense to send in the whole `type`? So we don't have to change this and add a new field if we happen to need another field from the `type`?

Reply:
I guess so 🤔

Reply:
It would also align with the other fields passed in already (`document` and `parent`). 👍 I'll approve given these changes (if you agree, of course).

Reply:
ok! Will change it